producer/contentProducer.go

```diff
@@ -0,0 +1,116 @@
+package producer
+
+import (
+	"encoding/json"
+	"errors"
+	"github.com/Financial-Times/message-queue-go-producer/producer"
+	"github.com/Financial-Times/uuid-utils-go"
+	log "github.com/Sirupsen/logrus"
+	gouuid "github.com/satori/go.uuid"
+)
+
+const (
+	uriBase             = "http://content-collection-unfolder.svc.ft.com/content/"
+	cmsContentPublished = "cms-content-published"
+	methodeSystemOrigin = "http://cmdb.ft.com/systems/methode-web-pub"
+)
+
+type ContentProducer interface {
+	Send(tid string, lastModified string, contentArr []map[string]interface{})
+}
+
+type defaultContentProducer struct {
+	msgProducer producer.MessageProducer
+}
+
+func NewContentProducer(msgProducer producer.MessageProducer) ContentProducer {
+	return &defaultContentProducer{
+		msgProducer: msgProducer,
+	}
+}
+
+func (p *defaultContentProducer) Send(tid string, lastModified string, contentArr []map[string]interface{}) {
+	for _, content := range contentArr {
+		p.sendSingleMessage(tid, content, lastModified)
+	}
+}
+
+func (p *defaultContentProducer) sendSingleMessage(tid string, content map[string]interface{}, lastModified string) {
+	logEntry := log.WithField("tid", tid)
+	uuid, err := p.extractUuid(content)
+	if err != nil {
+		logEntry.Warnf("Skip creation of kafka message. Reason: %v", err)
+		return
+	}
+
+	logEntry = logEntry.WithField("uuid", uuid)
+	msg, err := p.buildMessage(tid, uuid, lastModified, content)
+	if err != nil {
+		logEntry.Warnf("Skip creation of kafka message. Reason: %v", err)
+		return
+	}
+
+	err = p.msgProducer.SendMessage("", *msg)
+	if err != nil {
+		logEntry.Warnf("Unable to send message to Kafka. Reason: %v", err)
+	}
+}
+
+func (p *defaultContentProducer) extractUuid(content map[string]interface{}) (string, error) {
+	val, ok := content["uuid"]
+	if !ok {
+		return "", errors.New("No UUID found in content")
+	}
+
+	uuid, ok := val.(string)
+	if !ok {
+		return "", errors.New("Found UUID was not a string")
+	}
+
+	err := uuidutils.ValidateUUID(uuid)
+	if err != nil {
+		return "", err
+	}
+
+	return uuid, nil
+}
+
+func (p *defaultContentProducer) buildMessage(tid string, uuid string, lastModified string, content map[string]interface{}) (*producer.Message, error) {
+	body := publicationMessageBody{
+		ContentURI:   uriBase + uuid,
+		LastModified: lastModified,
+		Payload:      content,
+	}
+	bodyAsString, err := p.marshallToString(&body)
+	if err != nil {
+		return nil, err
+	}
+
+	headers := map[string]string{
+		"X-Request-Id":      tid,
+		"Message-Timestamp": lastModified,
+		"Message-Id":        gouuid.NewV4().String(),
+		"Message-Type":      cmsContentPublished,
+		"Origin-System-Id":  methodeSystemOrigin,
+		"Content-Type":      "application/json",
+	}
+
+	return &producer.Message{Headers: headers, Body: *bodyAsString}, nil
+
+}
+
+func (p *defaultContentProducer) marshallToString(body *publicationMessageBody) (*string, error) {
```

Comment:
I know. But I don't wan't this function to be usable without a `contentProducer`, even from within the package. In java I'd make this a private method. Since I cannot do that, I'll made it not exported and attached a receiver to it.  I think this is the most restrictive you can get with visibility in go.

If there is something even more restrictive, pls let me know.